src/wasm/wasm-type.cpp

```diff
@@ -196,42 +196,57 @@ struct TypeBounder {
   std::optional<Rtt> lub(const Rtt& a, const Rtt& b);
 };
 
-// Helper for printing types without infinitely recursing on recursive types.
+// Helper for generating names for HeapTypes when an alternative generator is
+// not available.
+struct DefaultHeapTypeNameGenerator {
+  size_t funcCount = 0;
+  size_t structCount = 0;
+  size_t arrayCount = 0;
+
+  void getName(std::ostream& os, HeapType type);
+};
+
+// Helper for printing types.
 struct TypePrinter {
-  size_t currDepth = 0;
-  std::unordered_map<TypeID, size_t> depths;
+  // Whether to print explicit supertypes.
+  bool hasSupertype;
```

Comment:
This confuses me. This is a TypePrinter for multiple types, while `hasSupertype` sounds like a property of a single type?

Reply:
I'll change this to `printSupertypes` to make it more clear.